vmngclient/api/partition_manager_api.py

```diff
@@ -30,69 +25,88 @@ class PartitionManagerAPI:
+        """
+        Set defualt software versions for devices
+
+        Args:
+            devices (DataSequence[Device]): devices
+            partition (Optional[str], optional): If none, current partition will be set as default,
+                else selected partition will be set as default
 
-    def set_default_partition(self, payload_devices: DataSequence[DeviceVersionPayload]) -> str:
+        Returns:
+            str: set partition task id
+        """
+        if partition:
+            payload_devices = self.device_version.get_device_available(partition, devices)
+        else:
+            payload_devices = self.device_version.get_devices_current_version(devices)
 
         url = "/dataservice/device/action/defaultpartition"
         payload = {
             "action": "defaultpartition",
             "devices": [asdict(device) for device in payload_devices],  # type: ignore
-            "deviceType": self.device_type.value,
+            "deviceType": get_install_specification(devices.first()).device_type.value,
         }
         set_default = dict(self.session.post(url, json=payload).json())
         return set_default["id"]
 
-    def remove_partition(self, devices_payload: DataSequence[DeviceVersionPayload], force: bool = False) -> str:
+    def remove_partition(
+        self, devices: DataSequence[Device], partition: Optional[str] = None, force: bool = False
+    ) -> str:
         """
-        Remove chosen software version from Vmanage repository
+        Remove chosen software version from device
 
         Args:
-            devices (List[Device]): remove partition for those devices
-            version (str): software version to be removed from repository
+            devices (DataSequence[Device]): remove partition for those devices
+            partition (str): If none, all availables partitions will be removed,
+                else selected partition will be removed
             force (bool): bypass version checks
 
         Returns:
             str: action id
         """
+        if partition:
+            payload_devices = self.device_version.get_device_available(partition, devices)
+        else:
+            payload_devices = self.device_version.get_devices_available_versions(devices)
 
         remove_partition_payload = [
-            RemovePartitionPayload(device.deviceId, device.deviceIP, device.version) for device in devices_payload
+            RemovePartitionPayload(device.deviceId, device.deviceIP, device.version)  # type: ignore
+            for device in payload_devices
         ]
+
         url = "/dataservice/device/action/removepartition"
         payload = {
             "action": "removepartition",
             "devices": [asdict(device) for device in remove_partition_payload],  # type: ignore
-            "deviceType": self.device_type.value,
+            "deviceType": get_install_specification(devices.first()).device_type.value,
```

Comment:
Should be there a validation whether there are not mixed devices type?

Reply:
Quite good idea, I thought about it. I suppose that in most cases, payload would be generated by filter(), in how many cases different personalities would be returned?

Reply:
We cannot limit users to use only `.filter()` method with our upgrades. 
A simple one-liner would work:
```python
len(set(device.personality for device in devices)) == 1
```
Also we need to add a proper message.
Anything else which is similar would work also.